policy/policy.go

```diff
@@ -156,28 +241,29 @@ func NewPoliciesFromSchema(policies []*schema.Policy) ([]Policy, error) {
 // policies by resolution in ascending order, finest resolution first.
 // If two policies have the same resolution, the one with longer
 // retention period comes first.
-type ByResolutionAsc []Policy
+type ByResolutionAsc []UnaggregatedPolicy
 
 func (pr ByResolutionAsc) Len() int      { return len(pr) }
 func (pr ByResolutionAsc) Swap(i, j int) { pr[i], pr[j] = pr[j], pr[i] }
 
 func (pr ByResolutionAsc) Less(i, j int) bool {
-	w1, w2 := pr[i].Resolution().Window, pr[j].Resolution().Window
+	pi, pj := pr[i].Policy(), pr[j].Policy()
+	w1, w2 := pi.Resolution().Window, pj.Resolution().Window
```

Comment:
We should be consistent here: either use pi and wi, or p1 and w1, but not mix them together.